lib/agenda/job-processing-queue.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @class
+ * @param {Object} args - Job Options
+ * @property {Object} agenda - The Agenda instance
+ * @property {Object} attrs
+ */
+class JobProcessingQueue {
+  constructor() {
+    this._queue = [];
+  }
+
+  get length() {
+    return this._queue.length;
+  }
+}
+
+/**
+ * Pops and returns last queue element (next job to be processed) without checking concurrency.
+ * @returns {Job} Next Job to be processed
+ */
+JobProcessingQueue.prototype.pop = function() {
+  return this._queue.pop();
+};
+
+/**
+ * Inserts job in first queue position
+ * @param {Job} job job to add to queue
+ * @returns {undefined}
+ */
+JobProcessingQueue.prototype.push = function(job) {
+  this._queue.push(job);
+};
+
+/**
+ * Inserts job in queue where it will be order from left to right in decreasing
+ * order of nextRunAt and priority (in case of same nextRunAt), if all values
+ * are even the first jobs to be introduced will have priority
+ * @param {Job} job job to add to queue
+ * @returns {undefined}
+ */
+JobProcessingQueue.prototype.insert = function(job) {
+  let matchIndex;
+  this._queue.some((element, index) => {
```

Comment:
Since node.js 4 can be simplified to:
```js
const matchIndex = this._queue.findIndex(element => {
    if (element.attrs.nextRunAt.getTime() <= job.attrs.nextRunAt.getTime()) {
      if (element.attrs.nextRunAt.getTime() === job.attrs.nextRunAt.getTime()) {
        if (element.attrs.priority >= job.attrs.priority) {
          return true;
        }
      } else {
        return true;
      }
    }
    return false;
  });
```

Reply:
Thank you! Your changes are in now. One of the test cases I was trying to fix failed now for Node 10 so I'll have a look again and try to fix it properly.